From c6b0f2c747a23a37323b26f9b31ed036fee03792 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Mon, 9 May 2016 14:45:14 +0800
Subject: [PATCH 1050/5242] MLK-12765 ARM: imx: make sure DLL is locked on
 i.MX7D

commit  51e620c107c0f52ab5921e94a4e548f3f7ba367f from
https://source.codeaurora.org/external/imx/linux-imx.git

On i.MX7D, per design team's require, need to make sure
DLL is locked after DDR frequency scaled done. Although
normally there should be no issue, but it is better to
add it.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
(cherry picked from commit 07c9f26b19b3ce05719d5634f1f56790b7ecf6af)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/ddr3_freq_imx7d.S |    2 +-
 arch/arm/mach-imx/lpddr3_freq_imx.S |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-imx/ddr3_freq_imx7d.S b/arch/arm/mach-imx/ddr3_freq_imx7d.S
index 843e181..85949d4 100644
--- a/arch/arm/mach-imx/ddr3_freq_imx7d.S
+++ b/arch/arm/mach-imx/ddr3_freq_imx7d.S
@@ -286,7 +286,7 @@
 	ldr	r8, =0x9880
 	str	r7, [r2, r8]
 
-	ldr	r7, =0x02020070
+	ldr	r7, =0x1010007e
 	str	r7, [r5, #DDRPHY_MDLL_CON0]
 
 	ldr	r7, =0x10000008
diff --git a/arch/arm/mach-imx/lpddr3_freq_imx.S b/arch/arm/mach-imx/lpddr3_freq_imx.S
index fbf3cb0..5c8c501 100644
--- a/arch/arm/mach-imx/lpddr3_freq_imx.S
+++ b/arch/arm/mach-imx/lpddr3_freq_imx.S
@@ -221,7 +221,7 @@
 	ldr	r8, =0x9880
 	str	r7, [r2, r8]
 
-	ldr	r7, =0x02020070
+	ldr	r7, =0x1010007e
 	str	r7, [r5, #DDRPHY_MDLL_CON0]
 
 	ldr	r7, =0x10000008
-- 
1.7.9.5

